jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -718,12 +720,14 @@ ecma_builtin_typedarray_prototype_set (ecma_value_t this_arg, /**< this argument
     }
 
     ecma_free_value (elem);
-
-    if (ecma_arraybuffer_is_detached (arraybuffer_p))
+    if (ecma_object_class_is (arraybuffer_p, ECMA_OBJECT_CLASS_ARRAY_BUFFER))
```

Comment:
Use && instead of `if`-s:
```
if (ecma_object_class_is (arraybuffer_p, ECMA_OBJECT_CLASS_ARRAY_BUFFER))
   && ecma_arraybuffer_is_detached (arraybuffer_p)
```
I had a question before: since sharedarraybuffer is never detached, `ecma_arraybuffer_is_detached` should always return false, doesn't it? In this case this code can be kept unchanged.

There are similar codes below.

Reply:
Yes it should but i forget to remove the `if`-s